src/core/lang.mli

```diff
@@ -108,24 +108,26 @@ val add_builtin :
   ?flags:Doc.Value.flag list ->
   ?meth:(string * Liquidsoap_lang.Type.scheme * string * value) list ->
   ?examples:string list ->
+  ?base:module_name ->
   string ->
   proto ->
   t ->
   (env -> value) ->
-  unit
+  module_name
```

Comment:
I don't really see the point in returning the module name here. Could you explain?

Reply:
This how you track dependencies. Module names are abstract types so, if you want to, say, register module `foo.bar`, you have to:
```ocaml
let foo = Lang.add_module "foo"
let foo_bar = Lang.add_module ~base:foo "bar
```
Same thing with operators:
```ocaml
let blank = Lang.add_operator "blank" ...

let blank_detect = Lang.add_operator ~base:blank "detect"
```

This way, if you have to extend a given record, you need to fetch its base/root from the module that defines it, thus creating the module dependencies required to order them at compilation time.